AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift

```diff
@@ -124,6 +129,41 @@ extension AWSDataStorePlugin: DataStoreBaseBehavior {
                              predicate: predicate(),
                              completion: onCompletion)
     }
+
+    public func clear(completion: @escaping DataStoreCallback<Void>) {
+        if storageEngine == nil {
+            completion(.successfulVoid)
+            return
+        }
+        storageEngine.clear { result in
+            var sError: Error?
+
+            switch result {
+            case .success:
+                self.log.verbose("Successfully called clear() on storage engine")
+            case .failure(let err):
+                sError = err
+                self.log.error("Failed to clear() storage engine")
+            }
+
+            //If we failed to kill the storage engine, we should
+            // continue to attempt to kill the storage engine & publishers
+            self.storageEngine = nil
+            if #available(iOS 13.0, *) {
+                if let publisher = self.dataStorePublisher as? DataStorePublisher {
+                    publisher.sendFinished()
+                }
+            }
+            self.dataStorePublisher = nil
+
+            if let error = sError {
```

Comment:
Why not just invoke `completion(result)`? You should then be able to get rid of L141-147 as well

Reply:
WHOA!  How did i not see that?!  Thank you!

updated to:
```
            self.storageEngine = nil
            if #available(iOS 13.0, *) {
                if let publisher = self.dataStorePublisher as? DataStorePublisher {
                    publisher.sendFinished()
                }
            }
            self.dataStorePublisher = nil
            completion(result)
```